### PR TITLE
Fixes for sumero contracts

### PR DESCRIPTION
**Pull request by ShubhamChndrvnshi**
Includes new sumero contracts, type changes and logic changes as per new contract

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [x]  Untested

### CODE REVIEW

**Comment by ShubhamChndrvnshi** [comment]
There are a lot of changes introduced lately in the repo, causing many test cases to fail.
Refer https://github.com/SumeroApp/uma-protocol/pull/18

**Comment by KohliSuraj** [comment]
Testing Liquidator Bot

Test sponsor wallet address: `0xaC7Bba69a23B32D5F0Db30E24143bc8a660aA2dc`
Liquidator bot wallet address: `0x6d8eb771f6503fd4db08e2879f582e6030c9b877`
cVT synth EMP: `0xCA23953A941044881F14b08F3BEDdc227E4928F2`

- Sponsor creates a slow withdrawal of 500 USDC, which should make the CR ratio below liquidation CR % of 120 %.
https://goerli.etherscan.io/tx/0x5d3039a85a53dfb9cbf81072575c41e9cb19266437ef46b09e7ecd1dcf38e67f
request withdrawal 500 USDC

- Liquidator bot does partial liquidation, since it does not have enough synth. Liquidates enough so that sponsor reach minimum sponsor position of 4.5
https://goerli.etherscan.io/tx/0x338473d871c1fb8424537dd4b0fe2f23f5b579db0d88809e5b347cd61876ae52
Liquidated by bot

File: packages/common/src/ProviderUtils.ts [hunk 1]
```diff
@@ -119,7 +119,7 @@ function addMnemonicToProvider(
 ): HDWalletProvider {
   const wallet = new HDWalletProvider(mnemonic, provider, keyOffset, numKeys);
   // Sumero-fix: modifying the default polling delay of hdwallet to avoid unnecessary eth calls
-  if (process.env.POLLING_DELAY) {
+  if (Number(process.env.POLLING_DELAY)) {
```
Comment (KohliSuraj) [inline packages/common/src/ProviderUtils.ts:122]: how does this help?

**Comment by KohliSuraj** [comment]
Testing Disputer Bot

https://goerli.etherscan.io/tx/0xdd78071594cecd4406681f341ce1e276170b71cd6fb6528a8af4b8ce6621b559

Posts a Disputer bond 
and
Does a price request to OO -> https://testnet.oracle.umaproject.org/?transactionHash=0xdd78071594cecd4406681f341ce1e276170b71cd6fb6528a8af4b8ce6621b559&eventIndex=129

https://testnet.oracle.umaproject.org/?transactionHash=0x59eaf1f04b5a5eec6fad8e66534dc880d33da27fb25b0240406aa21645e26f50&eventIndex=224


Proposed a Price of 92.85000 for Dispute no. 1 - https://goerli.etherscan.io/tx/0xe8673a534179a77746ca5094900500d0bc902a682d8dc3ab483bb118f9031b47

Proposed a Price of 150 for Dispute no. 2 - https://goerli.etherscan.io/tx/0xc224daa5869a35bd38a45dbd1e4351cb0101aa94b787ae484b3bcc452a3c0f52

Settled Dispute no. 1 via OO UI - https://goerli.etherscan.io/tx/0xc8aa1389ebdbfd92c0d297d83de643514d35fc5ac5b718ae9d92f16c8fd8632e

Settle Dispute no. 2 via OO UI - https://goerli.etherscan.io/tx/0xd222eaf54183cc03d12c7ccdecbe3292f518717a4d563e1d510f006053b66a91

File: packages/common/src/ProviderUtils.ts (hunk 1 above)
Comment (ShubhamChndrvnshi) [reply packages/common/src/ProviderUtils.ts:122]: process.env.POLLING_DELAY returns string value.
Boolean("0") --> true
Boolean(0)    --> false

Comment (KohliSuraj) [reply packages/common/src/ProviderUtils.ts:122]: ah, so env was returning string, and was setting a polling delay of 0, and actually never updating the blockchain details. Thanks for explaining.

**Comment by KohliSuraj** [comment]
@ShubhamChndrvnshi PR is looking good 👍 .

Only thing I would want us to test again, is if the bots are calling withdrawRewards which internally calls smart contract withdrawLiquidation function.

in bots disputer.js 

    const disputedLiquidations = this.financialContractClient
      .getDisputedLiquidations()
      .filter((liquidation) => liquidation.disputer === disputerAddress);
disputedLiquidations was coming as an empty array, even though, there is a disputed liquidation pending, which is unresolved.

Last, when I was testing this, I had to manually call withdrawLiquidation on the EMP, in order to transition the EMP to next state.

**Comment by ShubhamChndrvnshi** [comment]
> @ShubhamChndrvnshi PR is looking good 👍 .
> 
> Only thing I would want us to test again, is if the bots are calling withdrawRewards which internally calls smart contract withdrawLiquidation function.
> 
> in bots disputer.js
> 
> ```
> const disputedLiquidations = this.financialContractClient
>   .getDisputedLiquidations()
>   .filter((liquidation) => liquidation.disputer === disputerAddress);
> ```
> 
> disputedLiquidations was coming as an empty array, even though, there is a disputed liquidation pending, which is unresolved.
> 
> Last, when I was testing this, I had to manually call withdrawLiquidation on the EMP, in order to transition the EMP to next state.

Since the bots has a filter over all liquidations which needs to be withdrawn that's why this was happening.
As per discussion we are going to add a flag in bot, so that we can configure bot at anytime. If the bot's need to call withdrawLiquidations for users.

**Review by KohliSuraj — APPROVED** [review]

**Pull request merged**